### ISSUE TITLE
Cannot sort Aligned.out.sam files

### ISSUE THREAD

**Issue by csijcs** [alexdobin/STAR#465]
I have aligned some data with STAR_2.6.0c, and as far as I can tell everything went well except that I cannot sort the resulting Aligned.out.sam files.  I get different errors with each file.  The parameters I used for alignment are:

```
STAR --genomeDir /data/csijcs/hg38/STAR_index \
	--readFilesCommand zcat \
	--readFilesIn ${dir}/${samp}/${samp}_R1_001.fastq.gz ${dir}/${samp}/${samp}_R2_001.fastq.gz \
	--twopassMode Basic \
	--outReadsUnmapped None \
	--chimSegmentMin 12 \
	--chimJunctionOverhangMin 12 \
	--alignSJDBoverhangMin 10 \
	--alignMatesGapMax 100000 \
	--alignIntronMax 100000 \
	--chimSegmentReadGapMax 3 \
	--alignSJstitchMismatchNmax 5 -1 5 5 \
	--runThreadN 24 \
	--quantMode TranscriptomeSAM GeneCounts \
	--outSAMstrandField intronMotif \
	--outSAMattributes NH HI AS NM MD \
	--outSAMunmapped Within \
	--outFilterType BySJout
```

I had to remove the --outSAMtype BAM SortedByCoordinate parameter because I kept getting an error due to the --runThreadN 24, and chose to stay with the higher threads and sort after.  Now when I try to sort using the following:

`samtools sort Aligned.out.sam > Aligned.sortedByCoord.out.bam`

I get different errors for different Aligned.out.sam files, including:

```
[E::sam_parse1] SEQ and QUAL are of different length
[W::sam_read1] Parse error at line 202
[main_samview] truncated file.
```
```
[E::sam_parse1] unrecognized type 4
[W::sam_read1] Parse error at line 148808114
samtools sort: truncated file. Aborting
```
```
[E::sam_parse1] unrecognized type N
[W::sam_read1] Parse error at line 4888
samtools sort: truncated file. Aborting
```

Any idea what is going on here, and what I can do to fix this?

Many thanks

**Comment by alexdobin** [comment]
Hi @csijcs 

you have to convert SAM into BAM first:
$ samtools view -bS Aligned.out.sam > Aligned.out.bam
$ samtools sort Aligned.out.bam Aligned.sorted

Note that you can get the unsorted BAM (instead of SAM) file directly from STAR:
--outSAMtype BAM Unsorted

What was the error when you tried to sort within STAR?

Cheers
Alex

**Comment by csijcs** [comment]
Hi Alex,

I tried converting first and got the same issue.  I have sorted and converted directly in the past without issue.  I tried to recreate the error I got with the BAM sorting and it didn't make sense. I don't remember what it was, I just serched for the error and it semed that others had the same issue because the number of threads specified for STAR was more than the sorting could handle.  In the end I just reduded the number of processing threads to 12 and it worked fine, though it is mush slower at alignment.  Is there a maximum number of threads that can be used when sorting the BAM?  I have to process some more samples and it would be good to have more than 12 threads running. 24 would be good, but that's where I had problems before
